src/module_2/explicacion_data_analysis.md

```diff
@@ -0,0 +1,425 @@
+```python
+import pandas as pd
+import numpy as np
+import seaborn as sns
+import matplotlib.pyplot as plt
+```
+
+
+```python
+def check_product_id(product_id, dataframe):
+    return 1 if product_id in dataframe['variant_id'].values else 0
+```
+
+
+```python
+def make_corr_matrix_and_statictics(dataframe):
+ 
+ dataframe_numeric = dataframe.select_dtypes(include=[np.number])
+ corr_matrix = dataframe_numeric.corr()
+ #  para obtener las estadísticas 
+ print(dataframe_numeric.describe())
+ # Crear un heatmap de las correlaciones
+ sns.heatmap(corr_matrix, annot=True)
+ plt.show()
+```
+
+
+```python
+def freq_encoding_vendor_product_type(dataframe):
+# Calcular las frecuencias de aparición de cada categoría en 'vendor'
+ vendor_freq = df_orders_combined_inventory['vendor'].value_counts() / len(df_orders_combined_inventory)
+
+ # Calcular las frecuencias de aparición de cada categoría en 'product_type'
+ product_type_freq = df_orders_combined_inventory['product_type'].value_counts() / len(df_orders_combined_inventory)
+
+ # Mapear las frecuencias al DataFrame original
+ df_orders_combined_inventory['vendor_freq'] = df_orders_combined_inventory['vendor'].map(vendor_freq)
+ df_orders_combined_inventory['product_type_freq'] = df_orders_combined_inventory['product_type'].map(product_type_freq)
+
+ # Ahora, df tiene dos nuevas columnas: 'vendor_freq' y 'product_type_freq' con la frecuencia codificada
+```
+
+
+```python
+ 
+def convert_unix_to_readable(df, column_name):
+    df[column_name] = pd.to_datetime(df[column_name], unit='ms')
+    return df
+```
+
+
+```python
+def hours_vs_orders_plot(dataframe,hours_feature_name):
+ "contrastar diferentes horas con numero de oders"
+
+ dataframe['hour_of_day'] = dataframe[hours_feature_name].dt.hour
+ purchases_by_hour = dataframe.groupby('hour_of_day').size()
+ df_purchases_by_hour = purchases_by_hour.reset_index(name='number_of_purchases')
+
+ plt.figure(figsize=(14, 7))
+ plt.bar(df_purchases_by_hour['hour_of_day'], df_purchases_by_hour['number_of_purchases'])
+ plt.xlabel('Hour of Day')
+ plt.ylabel('Number of Purchases')
+ plt.title('Relationship between Hour of Day and Number of Purchases')
+ plt.xticks(df_purchases_by_hour['hour_of_day'])
+ plt.grid(True)
+ plt.show() 
+```
+
+
+```python
+def dates_vs_orders_plot(dataframe,dates_feature_name):
```

Comment:
Are these methods meant to be reusable? If not, you are in a notebook for exploration there is no need to have them in a method, which normally is good but for exploration may be tedious to go b ack and forth changing the method and calling the plot down in the notebook until you are done writing it into the final version